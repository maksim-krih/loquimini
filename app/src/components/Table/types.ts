import { GridPager, GridSearch } from "../../services/types";

export interface IProps {
  onRow: (record: any) => any;
  columns: any[];
  data: any[];
  pager: GridPager;
  loading: boolean;
  setPager: (_: any) => void;
  getData: (_: any, sorter: any, search?: GridSearch) => void;
}