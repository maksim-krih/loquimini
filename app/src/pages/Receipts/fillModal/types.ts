export interface IProps {
  visible: boolean;
  receiptId: string;
  onCancel: () => void;
}