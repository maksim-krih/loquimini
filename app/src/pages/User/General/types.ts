export interface IProps {
  isCreate?: boolean;
}
