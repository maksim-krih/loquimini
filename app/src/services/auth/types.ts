import { User } from "../types";

export interface Login {
  email: string;
  password: string;
}

export interface LoginResponse {
  user: User;
}