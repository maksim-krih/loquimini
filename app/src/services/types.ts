import { HouseType, ReceiptStatus, ReceiptType } from "../enums";

export interface CredentialsInfo {
  firstName: string;
  lastName: string;
  id: string;
  email: string;
  roles: Array<string>;
}
 
export interface UserCredentials {
  credentialsInfo: CredentialsInfo;
  accessToken: AccessToken;
}

export interface AccessToken {
  token: string;
  expiresIn: number;
}

export interface IError {
  code: string;
  desription: string;
}

export interface GridFilter {
  operator: string;
  field: string;
  value: string;
  logic: string;
}

export interface GridPager {
  current: number;
  total: number;
  pageSize: number;
}

export interface GridSorter {
  field: string;
  order: string;
}

export interface GridSearch {
  fields: Array<string>;
  value: string;
}

export interface GridRequest {
  filters: Array<GridFilter>;
  pager: GridPager;
  sorter: Array<GridSorter>;
  search: GridSearch;
}

export interface GridResponse<T> {
  total: number;
  list: T[];
}

export interface User {
  firstName: string;
  lastName: string;
  id: string;
  email: string;
  roles: Array<string>;
}

export interface House {
  id: string;
  number: string;
  street: string;
  type: HouseType;
  userId?: string;
  info?: BuildingInfo;
  flats: Array<Flat>
}

export interface Flat {
  id: string;
  houseId: string;
  houseNumber: string;
  street: string;
  number: string;
  info: BuildingInfo;
  userId: string;
}

export interface BuildingInfo {
  area: number;
  defaultIndicators: Array<DefaultIndicator>;
}

export interface DefaultIndicator {
  type: ReceiptType;
  value: number;
}

export interface CreateUser {
  firstName: string;
  lastName: string;
  email: string;
  password: string;
  roles: Array<string>;
}

export interface CreateHouse {
  number: string;
  street: string;
  type: HouseType;
  userId?: string;
  info?: BuildingInfo;
  flats: Array<Flat>;
}

export interface Receipt {
  id: string;
  type: ReceiptType;
  status: ReceiptStatus;
  rate: number;
  oldIndicator: number;
  newIndictor: number;
  total: number;
  paid: number;
  debt: number;
  date: Date;
  house: House;
  flat: Flat;
  houseType: HouseType;
}

export interface FillReceipt {
  receiptId: string;
  newIndicator: number;
}

export interface PayReceipt {
  receiptId: string;
  value: number;
}

export interface DashboardInfo {
  totalSum: number;
  totalDebts: number;
  currentFilled: number;
  totalFilled: number;
  currentPaid: number;
  totalPaid: number;
}