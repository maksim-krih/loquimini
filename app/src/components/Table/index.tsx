import React, { createRef, FC, useRef, useState } from 'react';
import { IProps } from './types';
import { useStyles } from './styles';
import { Button, Input, Space, Table as AntTable, TablePaginationConfig } from 'antd';
import { SearchOutlined } from '@ant-design/icons';

const Table: FC<IProps> = (props: IProps) => {
  const { onRow, columns, data, pager, loading, setPager, getData } = props;
  const classes = useStyles();
  const searchInputRef = createRef<Input>();
  const [searchText, setSearchText] = useState("");
  const [searchedColumn, setSearchedColumn] = useState("");
  
  const handleTableChange = (pagination: TablePaginationConfig, filters: any, sorter: any) => {
    setPager({...pager, current: pagination.current! });
    getData({...pager, current: pagination.current!}, sorter);
  };

  const getColumnSearchProps = ( dataIndex: any ) => ({
    filterDropdown: ({ setSelectedKeys, selectedKeys, confirm, clearFilters }: any) => (
      <div style={{ padding: 8 }}>
        <Input
          ref={searchInputRef}
          placeholder={`Search ${dataIndex}`}
          value={selectedKeys[0]}
          onChange={e => setSelectedKeys(e.target.value ? [e.target.value] : [])}
          onPressEnter={() => handleSearch(selectedKeys, confirm, dataIndex)}
          style={{ marginBottom: 8, display: 'block' }}
        />
        <Space>
          <Button
            type="primary"
            onClick={() => handleSearch(selectedKeys, confirm, dataIndex)}
            icon={<SearchOutlined />}
            size="small"
            style={{ width: 90 }}
          >
            Search
          </Button>
          <Button onClick={() => handleReset(clearFilters)} size="small" style={{ width: 90 }}>
            Reset
          </Button>
        </Space>
      </div>
    ),
    filterIcon: (filtered: any) => <SearchOutlined style={{ color: filtered ? '#1890ff' : undefined }} />,
    onFilter: (value: any, record: any) =>
      record[dataIndex]
        ? record[dataIndex].toString().toLowerCase().includes(value.toLowerCase())
        : '',
    onFilterDropdownVisibleChange: (visible: any) => {
      if (visible) {
        setTimeout(() => searchInputRef.current!.select(), 100);
      }
    }
  });

  const handleSearch = (selectedKeys: any, confirm: any, dataIndex: any) => {
    confirm();
    getData({...pager, current: 1}, null, { fields: [dataIndex], value: selectedKeys[0] });
    setSearchText(selectedKeys[0]);
    setSearchedColumn(dataIndex);
  };

  const handleReset = (clearFilters: any) => {
    clearFilters();
    setSearchText('');
  };

  const columnsModel = columns.map(x => x.search ? {...x, ...getColumnSearchProps(x.dataIndex) } : x);

  return (
    <div className={classes.container}>
      <AntTable
        columns={columnsModel}
        rowKey={(record: any) => record.id}
        dataSource={data}
        pagination={pager}
        loading={loading}
        onChange={handleTableChange}
        onRow={onRow}
      />
    </div>
  );
};

export default Table;