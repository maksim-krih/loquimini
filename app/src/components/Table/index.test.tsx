import Table from ".";
import { shallow } from "enzyme";
import { IProps } from "./types";

const setUp = (props: IProps) => shallow(<Table {...props} />);

describe("Table", () => {
  it("should render Table Component", () => {
    const component = setUp({
      columns: [],
      data: [],
      getData: () => {},
      loading: false,
      onRow: () => {},
      pager: {
        current: 0,
        pageSize: 0,
        total: 0
      },
      setPager: () => {}
    });
    expect(component).toMatchSnapshot();
  });
});