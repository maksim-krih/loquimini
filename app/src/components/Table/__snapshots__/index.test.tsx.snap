// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table should render Table Component 1`] = `
<div
  className="container-0-2-1"
>
  <Table
    columns={Array []}
    dataSource={Array []}
    loading={false}
    onChange={[Function]}
    onRow={[Function]}
    pagination={
      Object {
        "current": 0,
        "pageSize": 0,
        "total": 0,
      }
    }
    rowKey={[Function]}
  />
</div>
`;
